examples/statemachine/src/cli/generator.ts

```diff
@@ -8,196 +8,200 @@ import fs from 'fs';
 import { CompositeGeneratorNode, NL, processGeneratorNode } from 'langium';
 import { State, Statemachine } from '../language-server/generated/ast';
 
-export class StatemachineGenerator {
-    private statemachine: Statemachine;
-    private fileName: string;
-    private destination: string;
-    private fileNode: CompositeGeneratorNode = new CompositeGeneratorNode();
-
-    constructor(statemachine: Statemachine, fileName: string, destination: string | undefined) {
-        this.statemachine = statemachine;
-
-        const fileNameSeq = fileName.replace(/\..*$/, '').replaceAll(/[.-]/g, '').split('/');
-        this.fileName = `${fileNameSeq.pop() ?? 'statemachine'}.cpp`;
-        this.destination = destination ?? `./${fileNameSeq.join('/')}/generated`;
-    }
+export function generateCpp(statemachine: Statemachine, fileName: string, destination: string | undefined): string {
+    const fileNameSeq = fileName.replace(/\..*$/, '').replaceAll(/[.-]/g, '').split('/');
```

Comment:
Why split on every `/`, if we just join it again in the next instruction? Also, this could fail due to different separators (see Windows `\`)

Reply:
`fileNameSeq.pop()` in the next instruction removes the last segment, so, we join not everything. However, definitely there are better ways to implement it. I'll fix it.